keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentConstraintRunner.kt

```diff
@@ -43,14 +44,14 @@ class EnvironmentConstraintRunner(
   fun checkEnvironment(
     envContext: EnvironmentContext
   ) {
-    val pendingVersionsToCheck: MutableSet<String> =
-      when (envContext.environment.constraints.anyStateful) {
-        true ->
-          repository
-            .pendingConstraintVersionsFor(envContext.deliveryConfig.name, envContext.environment.name)
-            .filter { envContext.versions.contains(it) }
-            .toMutableSet()
-        false -> mutableSetOf()
+    val pendingVersionsToCheck: MutableList<PublishedArtifact> =
```

Comment:
How does this now return a list of published artifacts? `getPendingArtifactVersions` returns a list of strings, and I don't think you changed that in this PR

Reply:
I did. That's the function I wrote the test for in https://github.com/spinnaker/keel/pull/1652/commits/c653644f2d8e68d445fd47f79977c902308d77c3

Reply:
P.S. All the functions that used to return lists of version strings now return lists of `PublishedArtifact`.